consensus/polybft/contractsapi/bindings-gen/main.go

```diff
@@ -0,0 +1,308 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"text/template"
+
+	gensc "github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi/artifact"
+	"github.com/umbracle/ethgo/abi"
+)
+
+const (
+	contractVariableName   = "%sContract"
+	contractStructName     = "%sContractImpl"
+	contractVariableFormat = "%s = &%s{Artifact: %s}"
+	abiTypeNameFormat      = "var %sABIType = abi.MustNewType(\"%s\")"
+	eventNameFormat        = "%sEvent"
+)
+
+func main() {
+	cases := []struct {
+		contractName string
+		artifact     *artifact.Artifact
+		methods      []string
+		events       []string
+	}{
+		{
+			"StateReceiver",
+			gensc.StateReceiver,
+			[]string{
+				"commit",
+				"execute",
+			},
+			[]string{
+				"StateSyncResult",
+				"NewCommitment",
+			},
+		},
+		{
+			"ChildValidatorSet",
+			gensc.ChildValidatorSet,
+			[]string{
+				"commitEpoch",
+			},
+			[]string{},
+		},
+		{
+			"StateSender",
+			gensc.StateSender,
+			[]string{
+				"syncState",
+			},
+			[]string{
+				"StateSynced",
+			},
+		},
+		{
+			"CheckpointManager",
+			gensc.CheckpointManager,
+			[]string{
+				"submit",
+			},
+			[]string{},
+		},
+	}
+
+	rr := render{}
+	res := []string{}
+
+	for _, c := range cases {
+		for _, method := range c.methods {
+			res = append(res, rr.GenMethod(c.contractName, c.artifact.Abi.Methods[method]))
+		}
+
+		for _, event := range c.events {
+			res = append(res, rr.GenEvent(c.contractName, c.artifact.Abi.Events[event]))
+		}
+	}
+
+	str := `// Code generated by scapi/gen. DO NOT EDIT.
+package contractsapi
+
+import (
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo/abi"
+	"github.com/umbracle/ethgo"
+)
+
+`
+	str += strings.Join(res, "\n")
+
+	output, err := format.Source([]byte(str))
```

Comment:
I think that it would be nice if  `cobra.Command` is used in this `main` and if panics are not raised but errors handled

Reply:
Not sure if we would get anything useful, since this is called manually, and through `MakeFile`.

Reply:
I agree with Goran, this is only a small generator, it does not have any problem if it panics, it only worsens our own DX and it should not happen that often.